src/PerfView/OtherSources/XMLStackSource.cs

```diff
@@ -8,6 +8,8 @@
 using System.Xml;
 using System.Diagnostics;
 
+using OptimizationTier = Microsoft.Diagnostics.Tracing.Parsers.Clr.OptimizationTier;
```

Comment:
For Linux perf traces, XML is exported in place of the etlx on Windows, such that the conversion and export to XML/etlx are done only once per trace file and reused thereafter (or until the exported temp file is deleted). So the exported XML/etlx has to contain all of the info from the original trace since it is used as the source for everything else thereafter. The goal here was to always include the optimization tier in the XML, and when loading it (which is what happens when opening a Linux perf trace), optimization tiers will be shown based on config or stack view chosen.

Reply:
Another option which would give the best of both of the above options is to read/write a tagged value before code address infos to indicate a code address info version, and use that to look for or skip optimization tiers for each code address info. I'll give it a try.

Reply:
That might work since I believe that if it fails to read the tagged value it will back the stream up.  Let me know if that causes a bunch of grief and we can re-evaluate.

On the XML, that makes sense.  Thanks for the explanation.

Reply:
Sorry responded to the wrong comment, I meant that for the other comment about tagging, will continue about that there